`TestWatch` is often timeout

## Flaky Test

### Which jobs are failing

```
panic: test timed out after 10m0s
running tests:
	TestWatch (6m14s)
```

### CI link
https://github.com/tikv/pd/actions/runs/4292184591/jobs/7478308565

### Reason for failure (if possible)


### Anything else

--- a/tests/client/client_test.go
+++ b/tests/client/client_test.go
@@ -1439,11 +1439,11 @@ func TestWatch(t *testing.T) {
 
 	key := "test"
 	wg := sync.WaitGroup{}
+	ch, err := client.Watch(ctx, []byte(key))
+	re.NoError(err)
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		ch, err := client.Watch(ctx, []byte(key))
-		re.NoError(err)
 		var events []*meta_storagepb.Event
 		for e := range ch {
 			events = append(events, e...)

